### ISSUE TITLE
Unable to deploy 'Office-Add-in-NodeJS-SSO' in production server

### ISSUE THREAD

**Issue by mokarom** [OfficeDev/Office-Add-in-samples#875]
> Note: This repo is only for questions related to its samples. If you have questions about how to use office.js or the Office developer platform, please post your question on https://stackoverflow.com. Tag your question with office-js or outlook-web-addins

**Question** 
How to deploy 'Office-Add-in-NodeJS-SSO' in production?

_Ask your question here._
I'm able to test [this sample](https://github.com/OfficeDev/Office-Add-in-samples/tree/main/Samples/auth/Office-Add-in-NodeJS-SSO) in development mode. When tried to deploy in production I am not sure how this can be done. 
I'm trying to deploy in my Local IIS (IIS (Windows 10, IIS 10) but unable to make it work. The error says 404.0 not found
![DeployExcelAddin](https://github.com/user-attachments/assets/cfede724-8cdd-4f30-b36d-edd1d5521cd2)

I followed several articles to deploy node.js application - 
1. Tried to copy the whole 'Complete' folder and deployed as a web site in IIS (Windows 10, IIS 10).
2. https://medium.com/@adarsh-d/deploy-node-js-application-on-iis-9703d5dfcaca

Please provide a guidance on how the sample can be deployed in Local IIS.

**Comment by AlexJerabek** [comment]
Hi @mokarom,

Thank you for reporting this issue. @davidchesnut, could you please take a look?

**Comment by davidchesnut** [comment]
Hi @mokarom,

 Have you tried the article: [Deploy a single sign-on (SSO) Office Add-in to Microsoft Azure App Service](https://learn.microsoft.com/office/dev/add-ins/publish/deploy-office-add-in-sso-to-azure)? You don't have to use Azure as in the article, but I would take a look at the following items. 
- You need to have your web server configured to run server-side code (not static web site.)
- Even if using IIS you need to start the node server "node ./bin/www" as shown in the package.json.
- You need to be sure you updated any references to localhost:3000 to use the IIS domain:port number (unless it is same).

Hope this helps,
David

**Comment by mokarom** [comment]
> Have you tried the article: [Deploy a single sign-on (SSO) Office Add-in to Microsoft Azure App Service](https://learn.microsoft.com/office/dev/add-ins/publish/deploy-office-add-in-sso-to-azure)? You don't have to use Azure as in the article, but I would take a look at the following items.

Yes I tried this article and deployed in Azure app service and also in local IIS following the steps provided in the article. 
  - When I run it in Excel (both desktop and web) I am able to login successfully.
  - After login, the following error is shown when Microsoft Graph API is called to retrieve user's profile data

![Yeoman Generator Sample Error](https://github.com/user-attachments/assets/917ef31e-a120-49b6-8490-1aee85152649)

Please note that the the error screencast is taken from local deployment. 

According to this [SO post]( https://stackoverflow.com/questions/71400867/office-js-sso-getuserdata-method-not-working-when-deployed) _'yeoman office generator' cannot be used as the starter project for a production add-in that uses SSO'_ and suggested to try [Office Add-ins Samples - auth](https://github.com/OfficeDev/Office-Add-in-samples/tree/main/Samples/auth).

As specified before, 'Office Add-ins Samples - auth' sample runs fine in development mode. But I'm unable to deploy it in local IIS.

**Comment by mokarom** [comment]
After trying this article ([Deploy a single sign-on (SSO) Office Add-in to Microsoft Azure App Service](https://learn.microsoft.com/office/dev/add-ins/publish/deploy-office-add-in-sso-to-azure) again it worked. 

It is unclear why it worked this time. One thing I noticed that though I updated the .env file with 'NODE_ENV=production', but as I checked the file in Azure deployment the variable remained as NODE_ENV=development which doesn't make any sense. After manually updating the environment variable It worked without the error it was showing previously. 

Thanks for your help.
